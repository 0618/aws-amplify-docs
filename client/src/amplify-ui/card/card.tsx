import {Component, h, Prop, Host} from "@stencil/core";
import {
  cardStyle,
  graphicStyle,
  detailsStyle,
  externalLinkGraphic,
  hostStyle,
} from "./card.style";
import {createVNode} from "../../utils/hyperscript";

@Component({tag: "amplify-card", shadow: false})
export class AmplifyCard {
  /*** if true, the thumbnail gets rendered to the left of the detail (not above) */
  @Prop() readonly vertical?: boolean;
  /*** url */
  @Prop() readonly url?: string;
  /*** link tag to use */
  @Prop() readonly containerTag: string = "a";
  /*** whether or not to show external link graphic */
  @Prop() readonly external?: boolean;

  render() {
    const vertical = !!this.vertical;
    return (
      <Host class={hostStyle}>
        {createVNode(
          this.containerTag,
          {href: this.url, ...(this.external ? {target: "_blank"} : {})},
          <div class={{[cardStyle]: true, vertical}}>
            <div class={{[graphicStyle]: true, vertical}}>
              <slot name="graphic" />
            </div>
            <div class={detailsStyle}>
              <slot name="heading" />
              <slot name="description" />
            </div>
            {this.external && (
              <img
                class={externalLinkGraphic}
                src={`/assets/external-link-black.svg`}
                alt="External link"
              />
            )}
          </div>,
        )}
      </Host>
    );
  }
}
