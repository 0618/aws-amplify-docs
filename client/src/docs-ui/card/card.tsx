import {Component, h, Prop, Host} from "@stencil/core";

@Component({tag: "docs-card", shadow: false})
export class DocsCard {
  /*** if true, the thumbnail gets rendered to the left of the detail (not above) */
  @Prop() readonly vertical?: boolean;
  /*** url */
  @Prop() readonly url?: string;

  render() {
    const {vertical, url} = this;
    return (
      <Host>
        <amplify-card containerTag="docs-internal-link" {...{vertical, url}}>
          <slot name="graphic" />
          <slot name="heading" />
          <slot name="description" />
        </amplify-card>
      </Host>
    );
  }
}
