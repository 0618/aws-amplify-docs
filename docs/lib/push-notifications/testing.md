---
title: Testing
description: description
---

<inline-fragment platform="js" src="~/lib/push-notifications/fragments/js/testing.md"></inline-fragment>
