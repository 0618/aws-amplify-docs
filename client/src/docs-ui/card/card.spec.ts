import {DocsCard} from "./card";
import {newSpecPage} from "@stencil/core/testing";

describe("docs-card", () => {
  describe("Component logic", () => {
    let card: DocsCard;
    beforeEach(() => (card = new DocsCard()));

    it("should init `vertical` as `undefined`", () =>
      expect(card.vertical).toBeUndefined());
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [DocsCard],
            html: `<docs-card />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
