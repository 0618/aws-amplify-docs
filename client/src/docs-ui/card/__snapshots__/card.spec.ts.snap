// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs-card Render logic should render 1`] = `
<docs-card>
  <!---->
  <amplify-card containertag="docs-internal-link">
  </amplify-card>
</docs-card>
`;
